### ISSUE TITLE
Make it simpler to write custom metrics! 

### ISSUE THREAD

**Issue by nectario** [tensorflow/tensorflow#38670]
Why is it so complicated to write a simple custom metric? I do not want to deal with tensors, just numpy arrays. Please make it simpler and ability to use only numpy arrays. This request is for Keras.

<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.1
- Are you willing to contribute it (Yes/No):

**Describe the feature and the current behavior/state.**

**Will this change the current api? How?**

**Who will benefit with this feature?**

**Any Other info.**

**Comment by pavithrasv** [comment]
@nectario If you have a simple metric that is Mean based then you can write a function that accepts labels, predictions and returns the metric value as shown below. 

```
def mae(y_true, y_pred):
    return tf.keras.backend.mean(tf.math.abs(y_pred - y_true), axis=-1)
```

For anything else you can subclass the metric class, please see `BinaryTruePositives` example here:
https://www.tensorflow.org/api_docs/python/tf/keras/metrics/Metric?version=nightly

**Comment by nectario** [comment]
I need to write a metric that outputs the average number of positive results. I am just amazed how convoluted and complicated it is to write a simple metric like this.

**Comment by nectario** [comment]
Subclassing Metric class is not clear at all. I think there must be a way to write custom metric code without resorting to low level tensorflow code.

**Comment by nectario** [comment]
This is what I have so far to calculate the direction accuracy:

`from tensorflow.keras.metrics import Metric
from tensorflow.python.ops import init_ops
from tensorflow.python.ops import math_ops
from tensorflow.python.ops import array_ops

import tensorflow

class DirectionAccuracy(Metric):

    def __init__(self, name="direction_accuracy", dtype=None, **kwargs):
        super(Metric, self).__init__(name=name, dtype=dtype, **kwargs)
        self.total = self.add_weight("total", initializer=init_ops.zeros_initializer)
        self.count = self.add_weight("count", initializer=init_ops.zeros_initializer)
        self.direction_matches = self.add_weight("direction_matches", initializer=init_ops.zeros_initializer)


    def update_state(self, y_true, y_pred, sample_weight=None):
        self.direction_matches = math_ops.multiply(y_true, y_pred)
        res = math_ops.count_nonzero(tensorflow.greater_equal(self.direction_matches, 0.))
        self.count.assign_add(res)
       

    def result(self):
      return self.count`

**Comment by nectario** [comment]
Even when I am observing the source code for other custom metrics it's impossible to understand what is going on.

**Comment by nectario** [comment]
I should not have to spend two days to get this to work and still not working. I am fuming to say the least. Something so simple yet so difficult to do in Tensorflow. This is an example of utmost failure in usability. Convoluted brains coming together and creating a convoluted framework.

**Comment by pavithrasv** [comment]
Hello @nectario, sorry this has been difficult for you. For avg number of positive results you should just have to write a simple function and pass that to the compile API. Like the categorical accuracy function here: 

https://github.com/tensorflow/tensorflow/blob/2b220dae89d64be05276018e171fcfefa95c1bc5/tensorflow/python/keras/metrics.py#L3250

You do not have to subclass metric. Subclassing is required only if it not mean based.

**Comment by aramakus** [comment]
Agree with topic Author, I find the entry level barrier for basic extensions to be much higher than other alternatives. Maybe because I am DS and not Software Dev.

I would like to write F1Score by building on Precision and Recall classes that are already there with support of Sparse y_true. However those are classes, creating their new instances with a set of thresholds via a functional approach seems inefficient.

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
